### ISSUE TITLE
IOS Background Notifications 'contentAvailable'

### ISSUE THREAD

**Issue by buymybm100** [activitree/meteor-push#37]
Hi Paul, 
Thanks so much for this package! I used your instructions and the example to get up and running.  I am able to send notifications and receive them in pop-ups on the Mobile (outside the App). However, the CordovaPush.on("notifications") handler is not being called.

I am using your userPushNotification() meteor method .. and have the 'contentAvailable' flag set to true.
Below is the  Final notification right before shoot out but content-available is not set.
....

        "payload": {
          "aps": {
            "alert": {
                  "title": "You have a new Request ",
                  "body": "This is a simple alert",
                  "launch-image": ""
            },
            "badge": 1,
            "sound": "note.caf",
            "data": {}
          }
        }

Seems 'content-available' is needed for background notifications to work on IOS.
ref: https://github.com/havesource/cordova-plugin-push/blob/master/docs/PAYLOAD.md#ios-behaviour

Have I missed something ? T

Best wishes,
Beemer

**Comment by buymybm100** [comment]
Seems to be the same issue reported here : https://github.com/activitree/meteor-push/issues/35
ie, An exception is thrown in setBadge()  in  "lib/client/cordova.js"

If the code in setBadge() is hashed out , I can get notifications  from the CordovaPush.push.on("notification", fn) handler.
The badge on the App icon is set correctly even with the code hashed out in setBadge().

ERROR: Unhandled Promise rejection: undefined is not a function (near '...this.once...') ; Zone: <root> ; Task: Promise.then ; Value: TypeError: undefined is not a function (near '...this.once...') setBadge@http://localhost:12760/__cordova/packages/activitree_push.js?hash=6790efaf59ddf737434821a3c8a75eced5662294:114:16
http://localhost:12760/__cordova/packages/activitree_push.js?hash=6790efaf59ddf737434821a3c8a75eced5662294:226:32

```
 setBadge (count)  {
   /*---- **hashed out  temporarily** ----
    this.once('ready', () => {
      if (/ios/i.test(device.platform)) {
        this.log('PushHandle.setBadge:', count)
        // xxx: at the moment only supported on iOS
        this.push.setApplicationIconBadgeNumber(() => {
          this.log('PushHandle.setBadge: was set to', count)
        }, (e) => {
          PushEventState.emit('error', {
            type: getService() + '.cordova',
            error: 'PushHandle.setBadge Error: ' + e.message
          })
        }, count)
      }
    })
   -------- */
  }
```

what does setBadge() do ?

Best wishes,
Beemer

**Comment by buymybm100** [comment]
no,  i just used the cordova plugins specified in the example directory.

re: "authorization notification" - do you mean the popup message on the Mobile that ask if you allow notifications ?
Yes, i get that. It comes up after the call to CordovaPush.Configure({})..  comes up about 2-3 seconds after the call.

**Comment by rioadrian** [comment]
Hi @buymybm100,
I have similar problem and able to redirect my page on notification receive when I change my payload. It's also mentioned in documentation : [phonegap-push-plugin ](https://github.com/phonegap/phonegap-plugin-push/blob/master/docs/PAYLOAD.md#ios-behaviour)

try this one for payload:

```
{
    "registration_ids": [
        “this is yoour device id”
    ],
    "notification": {
        "body": “Just check it out guys, it’s cheap yet qualified!”,
        "title": “Buy Our Product Now!,
        "subtitle": “Just Stylist“,
        "sound": "default",
        "tag": "1"
    },
     "data": {
        "redirectUrl": "/merchandise/merchandiseDetail/_this_is_product_id_x7gdyneidjhma”
    },
  "content_available": true  
}
```
I hope it can help.

Thanks & Regards
